go-controller/pkg/util/dpu_annotations_test.go

```diff
@@ -14,140 +9,111 @@ var _ = Describe("DPU Annotations test", func() {
 	Describe("DPUConnectionDetails", func() {
 		var cd DPUConnectionDetails
 		var annot map[string]string
-		t := GinkgoT()
 
 		BeforeEach(func() {
 			cd = DPUConnectionDetails{}
 			annot = make(map[string]string)
 		})
 
-		Context("FromPodAnnotation()", func() {
-			It("Is populated correctly from annotations", func() {
```

Comment:
I am missing a test like `Is populated correctly from annotations`, both for legacy single network annotations and new multi network annotations, both for status and details. None of the tests here are testing the unmarshaling that happens within the Marshal methods.